src/resolver/ResolvedDatabaseAccountResource.ts

```diff
@@ -0,0 +1,54 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { AzExtTreeItem, IActionContext, ICreateChildImplContext, TreeItemIconPath } from "@microsoft/vscode-azext-utils";
+import { AppResource, ResolvedAppResourceBase } from "../api";
+import { DocDBAccountTreeItemBase } from "../docdb/tree/DocDBAccountTreeItemBase";
+import { MongoAccountTreeItem } from "../mongo/tree/MongoAccountTreeItem";
+import { PostgresServerTreeItem } from "../postgres/tree/PostgresServerTreeItem";
+
+export class ResolvedDatabaseAccountResource implements ResolvedAppResourceBase {
+    public id: string;
+    public contextValuesToAdd?: string[] | undefined = [];
```

Comment:
If we're initializing here then it's guaranteed to have a value; we don't need the `?` / `| undefined`, and the `?`s later on.